### ISSUE TITLE
main.py overwrites the location checkpoint with the shape checkpoint (both training phases save to the same .pth path)

### ISSUE THREAD

**Issue by TAKOS2022** [chuchen2017/GeoNeuralRepresentation#2]
**Description**
Introduced in ca91331 ("Refactor main.py and related scripts").

In main.py, both the location and shape training calls pass the same save_model_path to train_geo2vec_model:
-  Location phase
`location_model, location_embedding, _ = train_geo2vec_model(
    ...
    save_model_path=save_file_name,
)`
- Shape phase — trains second using the same path
`shape_model, shape_embedding, _ = train_geo2vec_model(
    ...
    save_model_path=save_file_name,
)`

**Problem**
utils/train_utils.py::train_geo2vec_model saves the model whenever a new best validation loss is reached:
`torch.save(model.state_dict(), save_model_path)`
Because the shape model trains after the location model and both use the same save_model_path, the shape checkpoint silently overwrites the location checkpoint.

As a result, after a complete training run:
`data/NYC_total_data.pth`
contains only the shape model weights.
The trained location decoder checkpoint is lost.

**Evidence**
This can be confirmed by inspecting a checkpoint produced by the current training pipeline:
`
import torch
sd = torch.load(
    "data/NYC_total_data.pth",
    map_location="cpu"
)
print(sd["location_encoder.freq_bands"].shape[0])
`
Output: 8 
This matches: num_freqs_shape = 8
rather than: num_freqs_location = 16

confirming that the final .pth file contains the shape model checkpoint.

**Impact**

The .npy embedding files are not affected.

The embedding saves following each training phase already use distinct paths:
`save_file_name.replace(".pth", "_loc")`
 `save_file_name.replace(".pth", "_shp")
`

Therefore, the embedding tables for entities already seen during training are preserved correctly.

However, the .pth checkpoint is required to restore the trained model/decoder and perform inference on new, unseen geometries or locations.

Since the location checkpoint is overwritten, the trained location decoder cannot be restored after training.

Suggested Fix

Use distinct checkpoint paths for the location and shape models, following the same naming convention already used for the .npy embeddings.
- Location : 
`
location_model, location_embedding, _ = train_geo2vec_model(  ...
    save_model_path=save_file_name.replace(
        ".pth",
        "_loc.pth"
    ),
)`

- Shape phase
`
    save_model_path=save_file_name.replace(
      ".pth",
        "_shp.pth"
    ),
)
`

This would produce: data/NYC_total_data_loc.pth and data/NYC_total_data_shp.pth

**Expected Behavior**

After a complete training run, both trained models should remain available independently:
`NYC_total_data_loc.pth    # Location model checkpoint`
`NYC_total_data_shp.pth    # Shape model checkpoint`
`NYC_total_data_loc.npy    # Location embeddings`
`NYC_total_data_shp.npy    # Shape embeddings`

**Comment by chuchen2017** [comment]
Thank you for raising the problem, and fix it!
